src/commands/captureDiagnostics.ts

```diff
@@ -0,0 +1,209 @@
+
+        ctx.outputChannel.log(`Saved diagnostics to ${diagnosticsDir}`);
+
+        const showInFinderButton = `Show In ${showCommandType()}`;
+        const copyPath = "Copy Path to Clipboard";
+        const result = await vscode.window.showInformationMessage(
+            `Saved diagnostic logs to ${diagnosticsDir}`,
+            showInFinderButton,
+            copyPath
+        );
+        if (result === copyPath) {
+            vscode.env.clipboard.writeText(diagnosticsDir);
+        } else if (result === showInFinderButton) {
+            exec(showDirectoryCommand(diagnosticsDir), error => {
+                // Opening the explorer on windows returns an exit code of 1 despite opening successfully.
+                if (error && process.platform !== "win32") {
+                    vscode.window.showErrorMessage(
+                        `Failed to open ${showCommandType()}: ${error.message}`
+                    );
+                }
+            });
+        }
+    } catch (error) {
+        vscode.window.showErrorMessage(`Unable to capture diagnostic logs: ${error}`);
+    }
+}
+
+async function writeLogFile(dir: string, name: string, logs: string) {
+    if (logs.length === 0) {
+        return;
+    }
+    await fs.writeFile(path.join(dir, name), logs);
+}
+
+function extensionLogs(ctx: WorkspaceContext): string {
+    return ctx.outputChannel.logs.join("\n");
+}
+
+function environmentLogs(ctx: WorkspaceContext): string {
+    const environmentOutputChannel = new SwiftOutputChannel("Swift", false);
+    ctx.toolchain.logDiagnostics(environmentOutputChannel);
+    environmentOutputChannel.log("Extension Settings:");
+    environmentOutputChannel.log(
+        JSON.stringify(vscode.workspace.getConfiguration("swift"), null, 2)
+    );
+    return environmentOutputChannel.logs.join("\n");
+}
+
+function diagnosticLogs(): string {
+    const diagnosticToString = (diagnostic: vscode.Diagnostic) => {
+        return `${severityToString(diagnostic.severity)} - ${diagnostic.message} [Ln ${diagnostic.range.start.line}, Col ${diagnostic.range.start.character}]`;
+    };
+
+    return vscode.languages
+        .getDiagnostics()
+        .map(
+            ([uri, diagnostics]) => `${uri}\n\t${diagnostics.map(diagnosticToString).join("\n\t")}`
+        )
+        .join("\n");
+}
+
+function sourceKitLogs(ctx: WorkspaceContext) {
+    return (ctx.languageClientManager.languageClientOutputChannel?.logs ?? []).join("\n");
+}
+
+async function sourcekitDiagnose(ctx: WorkspaceContext, dir: string) {
+    const sourcekitDiagnosticDir = path.join(dir, "sourcekit-lsp");
+    await fs.mkdir(sourcekitDiagnosticDir);
+
+    const toolchainSourceKitLSP = ctx.toolchain.getToolchainExecutable("sourcekit-lsp");
+    const lspConfig = configuration.lsp;
+    const serverPathConfig = lspConfig.serverPath;
+    const serverPath = serverPathConfig.length > 0 ? serverPathConfig : toolchainSourceKitLSP;
```

Comment:
Could you also pass the toolchain that the user has selected via `swift.path` to `sourcekit-lsp diagnose`? That way we will diagnose using the same toolchain that the user is building with if they have a custom SourceKit-LSP server set.